### ISSUE TITLE
Reg resizing of aux masks

### ISSUE THREAD

**Issue by puneeshkhanna** [benihime91/SIIM-COVID19-DETECTION-KAGGLE#1]
Hi,

I have been trying to learn from your awesome code and trying to replicate some parts as part of a reseach activity on this dataset.

I have a dount regarding resizing of aux masks.
aux_mask = albu.resize(
            np.array(aux_mask, np.float32),
            math.ceil(self.cfg.INPUT.INPUT_SIZE / 16),
            math.ceil(self.cfg.INPUT.INPUT_SIZE / 16),
        )

Should we instead not resize to self.cfg.MODEL.SEG_HEAD.LIN_FTRS as that will be the output feature map of aux segmentation head ? Or am I missing something ?

**Comment by puneeshkhanna** [comment]
Also one more minor doubt that are the albu augmentations getting applied to aux masks iamges also ?

**Comment by benihime91** [comment]
Hi, thanks for you interest. 

```
I have a dount regarding resizing of aux masks.
aux_mask = albu.resize(
  np.array(aux_mask, np.float32),
  math.ceil(self.cfg.INPUT.INPUT_SIZE / 16),
  math.ceil(self.cfg.INPUT.INPUT_SIZE / 16)
)
```
Actually in our experiments we had hard coded the values of the auxiliary mask to always be `INPUT_SIZE / 16`. This will be the `H, W` dimension of both the input segmentation mask and output auxiliary mask. 

`self.cfg.MODEL.SEG_HEAD.LIN_FTRS` this refers to the number of channels of the features extracted from the 2D backbone. And in any case the both the final output and input auxiliary mask will be of dimension (`1, INPUT_SIZE / 16, INPUT_SIZE / 16`.)


```
Also one more minor doubt that are the albu augmentations getting applied to aux masks iamges also ?
```
Yes augmentations are applied to masks as well.

**Comment by puneeshkhanna** [comment]
Hmm I think still a bit confused; If you check your code of 2D backbone, final conv2D output size will be of size SEG_HEAD.LIN_FTRS. Let me double check once more.

**Comment by puneeshkhanna** [comment]
self.fc = nn.Conv2d(nc, n_out, 1, bias=False, padding=0) is the last layer of backbone and here n_out is actually SEG_HEAD.LIN_FTRS

**Comment by puneeshkhanna** [comment]
Thanks I got it.  Got confused by number of filters. So basically final output dimension will remain say (32, 32) if input size is (512, 512) and the way your custom backbone is designed, you are not manipulating the size or actually ensuring size will remain same in the end.

**Comment by benihime91** [comment]
Exactly !

**Comment by puneeshkhanna** [comment]
I have a very good hang of your classification code. Great work again.
I will try to print model summaries along with sizes. Hope there is a way in pytorch as in tensorflow. Not an expert in pytorch. Important is to understand the size coming out of efficientNet architectures (self.enet.feature_info.channels()[0]) which is fed to seg head.
